File: frontend/src/app/models/pivot.models.ts
export interface PivotResponse {
  weekStart: string;
  weekEnd: string;
  days: PivotDay[];
  testSuites: string[];
  rows: PivotRow[];
  warnings: PivotWarning[];
}

export interface PivotDay {
  date: string;
  label: string;
}

export interface PivotRow {
  project: string;
  product: string;
  version: string;
  cells: Record<string, PivotCellItem[]>;
}

export interface PivotCellItem {
  title: string;
  description: string;
  machine: string;
  image: string;
  test: string;
  testSuite: string;
  scheduledTime: string;
}

export interface PivotWarning {
  title: string;
  crontab: string;
  message: string;
}

File: backend/FastPivot.Api/Models/PivotModels.cs
namespace FastPivot.Api.Models;

public sealed record PivotResponse(
    DateOnly WeekStart,
    DateOnly WeekEnd,
    IReadOnlyList<PivotDay> Days,
    IReadOnlyList<string> TestSuites,
    IReadOnlyList<PivotRow> Rows,
    IReadOnlyList<PivotWarning> Warnings);

public sealed record PivotDay(DateOnly Date, string Label);

public sealed record PivotRow(
    string Project,
    string Product,
    string Version,
    IReadOnlyDictionary<string, IReadOnlyList<PivotCellItem>> Cells);

public sealed record PivotCellItem(
    string Title,
    string Description,
    string Machine,
    string Image,
    string Test,
    string TestSuite,
    TimeOnly ScheduledTime);

public sealed record PivotWarning(string Title, string Crontab, string Message);
